src/amo/components/Link/index.js

```diff
@@ -95,7 +99,9 @@ export class LinkBase extends React.Component {
     };
 
     if (typeof href === 'string') {
-      const linkHref = urlPrefix ? joinUrl.pathname(urlPrefix, href) : href;
+      const linkHref = this.shouldPrepend(href, urlPrefix)
```

Comment:
Whoops, I meant moving the whole thing to a shared function within `render()`:

```js
const createLinkDest(urlString) => {
  return urlPrefix && !urlString.startsWith(urlPrefix)
    ? joinUrl.pathname(urlPrefix, urlString) : urlString;
}

const linkHref = createLinkDest(href);
const linkTo = createLinkDest(to);
```

It's not a big deal but maintaining the duplicated ternary could get awkward in the future (just like it was awkward when you had to add prefix detection).

Reply:
Oops, not sure why I didn't do that. I'll post a cleanup patch to take care of it now.